lib/elixir/test/elixir/kernel/sigils_test.exs

```diff
@@ -121,6 +187,124 @@ bar) in ["foo\\\nbar", "foo\\\r\nbar"]
     assert ~W(Foo.Bar.Baz)a == [:"Foo.Bar.Baz"]
   end
 
+  test "sigil w/W warns on trailing comma at compile time, not runtime" do
```

Comment:
I think this could be simplified to:

```elixir
@compile {:no_warn_undefined, Kernel.SigilsTest.TrailingComma}

test "sigil w/W warns on trailing comma at compile time, not runtime" do
  for sigil <- ~w(w W),
      modifier <- ~w(a s c) do
    assert capture_err(fn ->
             Code.compile_string("""
             defmodule Kernel.SigilsTest.TrailingComma do
               def run, do: ~#{sigil}(foo, bar baz)#{modifier}
             end
             """)
           end) =~
             "The sigils ~w/~W do not allow trailing commas"

    assert capture_err(fn -> Kernel.SigilsTest.TrailingComma.run() end) == ""

    purge(Kernel.SigilsTest.TrailingComma)
  end
end

defp purge(module) do
  :code.purge(module)
  :code.delete(module)
end
```

(with the code above, all those `Kernel.SigilsTest.Runtime_*` modules would not be required)

Reply:
Looks good to me, I'll roll this in.